src/pkg/repository/repository.go

```diff
@@ -288,3 +302,45 @@ func (r repository) DeleteBackup(ctx context.Context, id model.StableID) error {
 
 	return sw.DeleteBackup(ctx, id)
 }
+
+// ---------------------------------------------------------------------------
+// Repository ID Model
+// ---------------------------------------------------------------------------
+
+// repositoryModel identifies the current repository
+type repositoryModel struct {
+	model.BaseModel
+}
+
+const (
+	repositoryID = "repository-id"
+	repoIDTag    = "repo-id"
+)
+
+// should only be called on init.
+func newRepoModel(repoID string) repositoryModel {
+	return repositoryModel{
+		BaseModel: model.BaseModel{
+			ID: repositoryID,
+			Tags: map[string]string{
+				repoIDTag: repoID,
+			},
```

Comment:
I think we should store the generated ID in `BaseModel.ID` as it normally would be stored and leverage `GetIDsForType` (see below) to lookup the model

```suggestion
			ID: repoID,
```